mender-convert-modify

```diff
@@ -95,30 +96,49 @@ mkdir -p work/rootfs
 sudo mount ${boot_part} work/boot
 sudo mount ${root_part} work/rootfs
 
-mkdir -p work/mender-deb/files
+mkdir -p work/deb-packages
 
 log_info "Installing Mender client and related files"
 
 deb_arch=$(probe_debian_arch_name)
-DEBIAN_REVISION="-1"
+
 if [ "${MENDER_CLIENT_VERSION}" = "latest" ]; then
-  DEBIAN_REVISION=""
+  deb_name=$(deb_from_repo_dist_get "work/deb-packages" ${MENDER_APT_REPO_URL} ${deb_arch} "stable" "mender-client")
+elif [ "${MENDER_CLIENT_VERSION}" = "master" ]; then
+  deb_name=$(deb_from_repo_dist_get "work/deb-packages" ${MENDER_APT_REPO_URL} ${deb_arch} "experimental" "mender-client")
+else
+  DEBIAN_REVISION="-1"
+  deb_name=$(deb_from_repo_pool_get "work/deb-packages" ${MENDER_APT_REPO_URL} ${deb_arch} "mender-client" "${MENDER_CLIENT_VERSION}${DEBIAN_REVISION}")
 fi
-deb_name="mender-client_${MENDER_CLIENT_VERSION}${DEBIAN_REVISION}_${deb_arch}.deb"
-run_and_log_cmd "wget -Nq ${MENDER_STORAGE_URL}/${MENDER_CLIENT_VERSION}/dist-packages/debian/${deb_arch}/${deb_name} -P work/mender-deb"
-
-cd work/mender-deb
-run_and_log_cmd "ar -xv ${deb_name}"
-run_and_log_cmd "sudo tar xJf data.tar.xz -C files"
-cd - > /dev/null 2>&1
 
-run_and_log_cmd "sudo rsync --archive --keep-dirlinks --verbose work/mender-deb/files/ work/rootfs/"
+deb_extract_package "work/deb-packages/${deb_name}" "work/rootfs/"
 
 if [ "${MENDER_ENABLE_SYSTEMD}" == "y" ]; then
   run_and_log_cmd "sudo ln -sf /lib/systemd/system/mender-client.service \
         work/rootfs/etc/systemd/system/multi-user.target.wants/mender-client.service"
 fi
 
+if [ "${MENDER_ADDONS_INSTALL}" = "y" ]; then
+
+  log_info "Installing Mender addons"
+
+  if [ "${MENDER_ADDON_SHELL_VERSION}" = "latest" ]; then
+    deb_name=$(deb_from_repo_dist_get "work/deb-packages" ${MENDER_APT_REPO_URL} ${deb_arch} "stable" "mender-shell")
+  elif [ "${MENDER_ADDON_SHELL_VERSION}" = "master" ]; then
+    deb_name=$(deb_from_repo_dist_get "work/deb-packages" ${MENDER_APT_REPO_URL} ${deb_arch} "experimental" "mender-shell")
+  else
+    DEBIAN_REVISION="-1"
+    deb_name=$(deb_from_repo_pool_get "work/deb-packages" ${MENDER_APT_REPO_URL} ${deb_arch} "mender-shell" "${MENDER_ADDON_SHELL_VERSION}${DEBIAN_REVISION}")
+  fi
+
+  deb_extract_package "work/deb-packages/${deb_name}" "work/rootfs/"
+
+  if [ "${MENDER_ENABLE_SYSTEMD}" == "y" ]; then
+    run_and_log_cmd "sudo ln -sf /lib/systemd/system/mender-shell.service \
+          work/rootfs/etc/systemd/system/multi-user.target.wants/mender-shell.service"
+  fi
```

Comment:
@merlin-northern This piece enables the service by default. Is this something that we don't want? Then we have two options:
* Have a separate variable `MENDER_ADDON_SHELL_ENABLE_SYSTEMD`
* Ignore it all together

I don't like any of the above, do we have other ideas?

Reply:
The service should *not* be enabled (as in: started) by default. At least that is what I understood from all the discussions and comments by Eystein (@estenberg). if you do not like a separate variable, we can safely let user to start it, and do not enable it by default.